Decisions.TruBot/Steps/WaitRunBotStep.cs

```diff
@@ -0,0 +1,120 @@
+using System.Net.Http.Json;
+using Decisions.TruBot.Api;
+using Decisions.TruBot.Data;
+using DecisionsFramework;
+using DecisionsFramework.Data.ORMapper;
+using DecisionsFramework.Design.ConfigurationStorage.Attributes;
+using DecisionsFramework.Design.Flow;
+using DecisionsFramework.Design.Flow.Mapping;
+using DecisionsFramework.Design.Flow.StepImplementations;
+using DecisionsFramework.Design.Properties;
+using DecisionsFramework.ServiceLayer;
+
+namespace Decisions.TruBot.Steps
+{
+    [Writable]
+    [AutoRegisterStep("Run Bot and Wait", "Integration/TruBot/Bot")]
+    [ShapeImageAndColorProvider(null, TruBotSettings.TRUBOT_IMAGES_PATH)]
+    public class WaitRunBotStep : IAsyncStep, IDataConsumer
+    {
+        private const string PATH_DONE = "Done";
+        private const string TRUBOT_RESPONSE = "TruBot Status Response";
+        
+        private const string BOT_ID = "BotId";
+        
+        [WritableValue]
+        private string overrideBaseUrl;
+
+        [PropertyClassification(0, "Override Base URL", "Override Settings")]
+        public string OverrideBaseUrl
+        {
+            get => overrideBaseUrl;
+            set => overrideBaseUrl = value;
+        }
+
+        public void Start(StepStartData data)
+        {
+            int botId = data.Data[BOT_ID] as int? ?? 0;
```

Comment:
If this field is marked as ignored in the input, 
does it actually come in as a null or an int?

nullable int is usually an odd pattern. 

Is `botId = 0` ever valid? or can we just check for this value and throw an exception if it doesn't parse to int in the first place?